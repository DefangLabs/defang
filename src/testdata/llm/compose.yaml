services:
  llm:
    x-defang-llm: true
    image: "llm:latest"

  gateway-with-ports:
    x-defang-llm: true
    image: "defang.io/openai-access-gateway:latest"
    ports:
      - 5678:5678

  gateway-without-ports:
    x-defang-llm: true
    image: "defang.io/openai-access-gateway:latest"

  not-our-gateway:
    x-defang-llm: true
    image: "fakerepo.com/openai-access-gateway:latest"