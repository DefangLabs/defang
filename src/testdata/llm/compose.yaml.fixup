{
  "alt-repo": {
    "command": null,
    "entrypoint": null,
    "image": "altrepo.com/openai-access-gateway:latest",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 80,
        "protocol": "tcp"
      }
    ]
  },
  "gateway-with-ports": {
    "command": null,
    "entrypoint": null,
    "image": "defang.io/openai-access-gateway:latest",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "ingress",
        "target": 5678,
        "published": "5678",
        "protocol": "tcp",
        "app_protocol": "http"
      }
    ]
  },
  "gateway-without-ports": {
    "command": null,
    "entrypoint": null,
    "image": "defang.io/openai-access-gateway:latest",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 80,
        "protocol": "tcp"
      }
    ]
  },
  "llm": {
    "command": null,
    "entrypoint": null,
    "image": "llm:latest",
    "networks": {
      "default": null
    }
  }
}