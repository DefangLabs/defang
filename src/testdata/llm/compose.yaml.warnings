 ! service "alt-repo": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "gateway-with-ports": ingress port without healthcheck defaults to GET / HTTP/1.1
 ! service "gateway-with-ports": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "gateway-without-ports": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "llm": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors