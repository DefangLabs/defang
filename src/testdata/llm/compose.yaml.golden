name: llm
services:
  alt-repo:
    image: altrepo.com/openai-access-gateway:latest
    networks:
      default: null
    x-defang-llm: true
  gateway-with-ports:
    image: defang.io/openai-access-gateway:latest
    networks:
      default: null
    ports:
      - mode: ingress
        target: 5678
        published: "5678"
        protocol: tcp
    x-defang-llm: true
  gateway-without-ports:
    image: defang.io/openai-access-gateway:latest
    networks:
      default: null
    x-defang-llm: true
  llm:
    image: llm:latest
    networks:
      default: null
    x-defang-llm: true
networks:
  default:
    name: llm_default
